### ISSUE TITLE
Executing runc as a SCHED_RR process fails

### ISSUE THREAD

**Issue by bgray** [opencontainers/runc#806]
On Linux when trying to start a container if runc itself is in SCHED_RR mode it fails, even it you set a realtime bandwidth in the config.json.

For example my config.json has the following

```
        "resources": {
            "cpu": {
                "realtimeRuntime": 10000,
                "realtimePeriod": 1000000
            }
            ....
        }
```

And I try and execute runc like this

```
chrt -r 6 runc start blah
```

Then I get the following error

```
failed to write 4531 to cgroup.procs: write /sys/fs/cgroup/cpu/blah/cgroup.procs: invalid argument
```

I believe the issue is that the `cpu.rt_runtime_us` cgroup value is not inherited and instead always defaults to 0.  And the code in `runc/libcontainer/cgroups/fs/cpu.go` moves the process into the container before setting the `cpu.rt_runtime_us` value, and therefore because the process has inherited the SCHED_RR policy from runc it can't be moved into the cgroup due to the cgroup having 0 rt bandwidth.

The solution that worked for me was to set the RT scheduling values in the cpu cgroup before moving in the process (like how the `memory.kmem.limit_in_bytes` is set in the memory cgroup before moving in the process).

I've attached a patch of my changes.
[sched-rr-cpu-cgroup-diff.txt](https://github.com/opencontainers/runc/files/252246/sched-rr-cpu-cgroup-diff.txt)

So is there any appetite for this change? or do people think you shouldn't be using SCHED_RR anyway. Happy to create a pull request if wanted.

**Comment by cyphar** [comment]
You can definitely open a PR for this change, if it means `runc` can work in even more setups.

**Comment by hqhq** [comment]
Fixed by #860 .
